doc/source/whatsnew/v2.1.0.rst

```diff
@@ -234,6 +234,7 @@ Deprecations
 - Deprecated :func:`is_datetime64tz_dtype`, check ``isinstance(dtype, pd.DatetimeTZDtype)`` instead (:issue:`52607`)
 - Deprecated unused "closed" and "normalize" keywords in the :class:`DatetimeIndex` constructor (:issue:`52628`)
 - Deprecated unused "closed" keyword in the :class:`TimedeltaIndex` constructor (:issue:`52628`)
+- Deprecated :func:`is_datetime64_any_dtype`, check for specific dtypes of interest instead (:issue:`52651`)
```

Comment:
i guess.  all i really care about is that we dont use this internally.  but i do think its a footgun since pretty much anyone who uses it wrote it with only numpy/pd.DatetimeTZDtype in mind